aspnetcore/blazor/javascript-interoperability/import-export-interop.md

```diff
@@ -0,0 +1,417 @@
+>
+> The conditional check for <xref:System.OperatingSystem.IsBrowser%2A?displayProperty=nameWithType> in the preceding example ensures that the code is only called in Blazor WebAssembly apps running on the client in a browser. This is important for library code deployed as NuGet packages that might be referenced by developers of Blazor Server apps, where the preceding code can't execute. If you know for sure that the code is only implemented in Blazor WebAssembly apps, you can remove the check for <xref:System.OperatingSystem.IsBrowser%2A?displayProperty=nameWithType>.
+
+To export a .NET method so that it can be called from JS, use the `[JSExport]` attribute.
+
+In the following example:
+
+* `SetWelcomeMessage` calls a JS function named `setMessage`. The JS function calls into .NET to receive the welcome message from `GetMessageFromDotnet` and displays the message in the UI.
+* `GetMessageFromDotnet` is a .NET method with the `[JSExport]` attribute that returns a welcome message, "Hello from Blazor!" in Portuguese.
+
+`Pages/CallDotNet1.razor.cs`:
+
+```csharp
+using System.Runtime.InteropServices.JavaScript;
+using System.Runtime.Versioning;
+
+namespace BlazorSample.Pages
+{
+    [SupportedOSPlatform("browser")]
+    public partial class CallDotNet1
+    {
+        [JSImport("setMessage", "CallDotNet1")]
+        internal static partial void SetWelcomeMessage();
+
+        [JSExport]
+        internal static string GetMessageFromDotnet()
+        {
+            return "Olá do Blazor!";
+        }
+    }
+}
+```
+
+The app's namespace for the preceding `CallDotNet1` partial class is `BlazorServer`. The component's namespace is `BlazorSample.Pages`. If using the preceding component in a local test app, update the app's namespace to match the app. For example, the component namespace is `ContosoApp.Pages` if the app's namespace is `ContosoApp`. For more information, see <xref:blazor/components/index#partial-class-support>.
+
+In the following example, a JS function named `setMessage` is imported from a collocated JS file.
+
+The `setMessage` method:
+
+* Calls `globalThis.getDotnetRuntime(0)` to expose the WebAssembly .NET runtime instance for calling exported .NET methods.
+* Obtains the app assembly's JS exports. The name of the app's assembly in the following example is `BlazorSample`.
+* Calls the `BlazorSample.Pages.CallDotNet1.GetMessageFromDotnet` method from the exports (`exports`). The returned value, which is the welcome message, is assigned to the `CallDotNet1` component's `<span>` text. The app's namespace is `BlazorSample`, and the `CallDotNet1` component's namespace is `BlazorSample.Pages`.
+
+`Pages/CallDotNet1.razor.js`:
+
+```javascript
+export async function setMessage() {
+  const { getAssemblyExports } = await globalThis.getDotnetRuntime(0);
+  var exports = await getAssemblyExports("BlazorSample.dll");
```

Comment:
In real world apps, you could probably put these two lines in a JS initializer (or modify the logic here to cache `exports`), since they only have to be executed once. But it might be fine to leave as-is in this example, for the sake of simplicity.